dagger/stores/fsstore.nim

```diff
@@ -0,0 +1,115 @@
+## Nim-Dagger
+## Copyright (c) 2021 Status Research & Development GmbH
+## Licensed under either of
+##  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+##  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+## at your option.
+## This file may not be copied, modified, or distributed except according to
+## those terms.
+
+{.push raises: [Defect].}
+
+import std/os
+
+import pkg/chronos
+import pkg/chronicles
+import pkg/libp2p
+import pkg/questionable
+import pkg/questionable/results
+import pkg/stew/io2
+
+import ./memorystore
+import ./blockstore
+import ../blocktype
+
+export blockstore
+
+logScope:
+  topics = "dagger fsstore"
+
+type
+  FSStore* = ref object of BlockStore
+    cache: BlockStore
+    repoDir: string
+    postfixLen*: int
+
+template blockPath(self: FSStore, cid: Cid): string =
+  self.repoDir / ($cid)[(($cid).len - self.postfixLen)..<($cid).len]
```

Comment:
```suggestion
proc blockPath(self: FSStore, cid: Cid): string =
  let sid = $cid
  self.repoDir / sid[^self.postfixLen..^1] / sid
```
And use blockPath.parentDir to get the directory when you want it

Less error prone, imo (eg, delBlock is wrong)